main.py

```diff
@@ -12,6 +13,13 @@
 if not os.path.exists(data_file):
 	save_csv_from_url(url, data_file)
 
+# Fetch data if it is more than one day out of date 
+today = date.today() 
+last_date = date.today() #pandas.read_csv(data_file, names=headers, usecols=[0], skiprows=[0:-2])
```

Comment:
You need to use: e.g. https://docs.python.org/3/library/os.path.html#os.path.getmtime on the file to check when it was last "modified" and compare to current time.